$OpenBSD: patch-qucs_CMakeLists_txt,v 1.2 2019/06/22 20:18:00 rsadowski Exp $

Fix build with Ninja 1.9.0.
Create only one custom target to generate config.h to avoid duplicate ninja
rules.

fix man install directory

Index: qucs/CMakeLists.txt
--- qucs/CMakeLists.txt.orig
+++ qucs/CMakeLists.txt
@@ -75,12 +75,6 @@ IF(NOT HAVE_QTABWIDGET_SETMOVABLE)
   MESSAGE("QTabWidget::setMovable() not available, feature disabled.")
 ENDIF()
 
-# configure the header config.h
-CONFIGURE_FILE (
-    "${PROJECT_SOURCE_DIR}/../config.h.cmake"
-    "${PROJECT_BINARY_DIR}/config.h"
-)
-
 INCLUDE_DIRECTORIES("${PROJECT_BINARY_DIR}")
 INCLUDE_DIRECTORIES( ${CMAKE_SOURCE_DIR} )
 
@@ -249,6 +243,7 @@ ADD_EXECUTABLE( ${QUCS_NAME} MACOSX_BUNDLE WIN32
 # Tell CMake which libraries we need to link our executable against.
 #
 TARGET_LINK_LIBRARIES( ${QUCS_NAME}  components diagrams dialogs paintings extsimkernels spicecomponents ${QT_LIBRARIES} )
+ADD_DEPENDENCIES(${QUCS_NAME} generate_config_header)
 
 #
 # Prepare the installation
@@ -318,7 +313,7 @@ ENDIF()
 
 INSTALL( FILES ${SCRIPTS} DESTINATION bin/ )
 
-INSTALL( FILES ${QUCS_NAME}.1 DESTINATION share/man/man1 )
+INSTALL( FILES ${QUCS_NAME}.1 DESTINATION man/man1 )
 
 # To Create a package, one can run "cpack -G DragNDrop CPackConfig.cmake" on Mac OS X
 # where CPackConfig.cmake is created by including CPack
